nimble/controller/src/ble_ll_xcvr.c

```diff
@@ -102,7 +102,10 @@ void
 ble_ll_xcvr_rfclk_timer_exp(void *arg)
 {
     if (g_ble_ll_data.ll_rfclk_state == BLE_RFCLK_STATE_OFF) {
+        os_sr_t sr;
```

Comment:
'os_sr_t sr' shall be on top of the function according to coding guide

This is timer callback function which in nimble is called when interrupts disabled, could you do the same for RIOT?

Reply:
this is called the same way in RIOT but since `OS_ASSERT_CRITICAL` does not check explicitly that interrupts are disabled (see https://github.com/apache/mynewt-nimble/pull/366) it will fail because in this case they were disabled by `__HAL_DISABLE_INTERRUPTS` (https://github.com/apache/mynewt-nimble/blob/master/porting/nimble/src/hal_timer.c#L28). I think a better solution would be to change those macros in `hal_timer.c` to use `ble_npl_hw_enter_critical` and `ble_npl_hw_exit_critical` so they "obey" hack in NPL OS implementation.

Reply:
+1

will propose an alternative to this PR taking the above comments into account. Together with #366 that should hopefully solve this for good.

Reply:
proposed a fix in https://github.com/apache/mynewt-nimble/pull/369